modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/ExecutorImpl.scala

```diff
@@ -3,66 +3,99 @@ package edu.gemini.seqexec.server
 import edu.gemini.seqexec.odb.SeqExecService
 import edu.gemini.spModel.core.Peer
 
-import scalaz._
+import scalaz.{\/, _}
 import Scalaz._
-
 import edu.gemini.pot.sp.SPObservationID
 import edu.gemini.seqexec.server.Executor._
 import edu.gemini.seqexec.server.SeqexecFailure._
 import edu.gemini.spModel.config2.ConfigSequence
 
 import scalaz.concurrent.Task
+import scalaz.stream.Process
+import scalaz.stream.async
 
 /** An executor that maintains state in a pair of `TaskRef`s. */
 class ExecutorImpl private (cancelRef: TaskRef[Set[SPObservationID]], stateRef: TaskRef[Map[SPObservationID, Executor.ExecState]]) {
+  type ExecutionFailures = NonEmptyList[SeqexecFailure] \/ Unit
+  type ExecutionResult = (ExecState, ExecutionFailures)
 
   private var loc = new Peer("localhost", 8443, null)
 
+  // Queue to keep events generated when executing a sequence
+  private val eventsQueue = async.topic[SeqexecEvent]()
```

Comment:
so, the trouble is with constructing the topic there instead of being a param, I guess the exact equivalence would be a constructor parameter with a default value?